server/protocols/webadmin/webadmin-core/src/main/java/org/apache/james/webadmin/routes/HealthCheckRoutes.java

```diff
@@ -101,11 +113,25 @@ public Object performHealthCheckForComponent(Request request, Response response)
     }
 
     public Object getHealthChecks(Request request, Response response) {
-        return healthChecks.stream()
-                .map(healthCheck -> new HealthCheckDto(healthCheck.componentName()))
-                .collect(ImmutableList.toImmutableList());
+        return healthCheckMap.values().stream()
+            .map(healthCheck -> new HealthCheckDto(healthCheck.componentName()))
+            .collect(ImmutableList.toImmutableList());
+    }
+
+    private Collection<HealthCheck> getHealthChecks(String componentNamesString) {
+        List<ComponentName> componentNames = Arrays.stream(componentNamesString.split("\\+")).map(ComponentName::new).toList();
```

Comment:
what happend if `component name` has "+" character?
IMO we can use a standard of http request
when check is an array
the syntax should be
`check=RabbitMQJmapEventBusDeadLetterQueueHealthCheck&check=OpenSearch%20Backend`

Reply:
> what happend if component name has "+" character?

I think the component can and likely should be URL encoded.

`OpenSearch+Backend` => `OpenSearch%2BBackend` ~ `checks=RabbitMQJmapEventBusDeadLetterQueueHealthCheck+OpenSearch%2BBackend` may be fine.

But it seems the `+` character has another special meaning in the URL:
![image](https://github.com/user-attachments/assets/a948a496-85e6-4835-8060-139f03319920)

cf: https://www.freecodecamp.org/news/url-encoded-characters-reference/#:~:text=There%20are%20only%20certain%20characters,that%20can%20have%20special%20meanings.

=> Likely we should avoid using the `+` as delimiter here as it does not mean a space.

Reply:
updated:  check=RabbitMQJmapEventBusDeadLetterQueueHealthCheck&check=OpenSearch%20Backend